kin-devplatform-sdk/src/main/java/kin/devplatform/data/order/OrderRepository.java

```diff
@@ -158,17 +185,32 @@ public void onResponse(Order response) {
 
 			@Override
 			public void onFailure(ApiException e) {
-				getOrderWatcher().postValue(
-					new Order().orderId(order.getId()).offerId(order.getOfferId()).status(Status.FAILED)
-						.error(e.getResponseBody()));
-				removeCachedOpenOrderByID(order.getId());
+				updateFailure(e.getResponseBody());
 				if (callback != null) {
 					callback.onFailure(ErrorUtil.fromApiException(e));
 				}
 			}
+
+			private void updateFailure(Error e) {
+				getOrderWatcher().postValue(
+					new Order().orderId(order.getId()).offerId(order.getOfferId()).status(Status.FAILED)
+						.error(e));
+				removeCachedOpenOrderByID(order.getId());
+			}
 		});
 	}
 
+	private void validateBlockchainVersions(BlockchainData blockchainData) throws MigrationNeededExceptions {
```

Comment:
can be shared with `CreateExternalOrderCall`.
why to throw and catch? you can return boolean instead and generate the Exception in the thrower clause

Reply:
Can't be shared but did what you suggested regarding the "throw and catch and boolean"